### PR TITLE
Chatbot project - project number 4- Pair programming done by Anna Kaisa Honkonen and Estefanía Quevedo

### PR DESCRIPTION
**Pull request by esteficodes**
Pair programming done by Anna Kaisa Honkonen and Estefanía Quevedo

### CODE REVIEW

File: code/script.js [hunk 1]
```diff
@@ -36,12 +44,183 @@ const greeting = () => {
...
+          <option id="select">Here comes the sun by the Beatles</option>
+          <option id="select">Shiny happy people by REM</option>
+        </select>
+      `
+      const select = document.getElementById("select");
+      let userName = document.getElementById('user-input');
+      select.addEventListener("change", () =>
+      greatChoice(select.value))
+    }
+  })
```
Comment (KarolCislowski) [inline code/script.js:91]: You already created an event listener for this event and DOM object. You should clean up it a little. 😸

File: code/script.js [hunk 2]
```diff
@@ -36,12 +44,183 @@ const greeting = () => {
...
+            <option id="">In the end by Linkin Park</option>
+            <option id="">Basket case by Green Day</option>
+            <option id="">Break Stuff by Limp Bizkit</option>
+         </select>
+        `
+        const select = document.getElementById("select");
+        select.addEventListener("change", () =>
+        greatChoice(select.value))
+      }   
+   })
```
Comment (KarolCislowski) [inline code/script.js:125]: all 3 events have the same skeleton ☠️ This is a great opportunity to extract this logic to the function and call it when needed. 😸

File: code/script.js [hunk 3]
```diff
@@ -36,12 +44,183 @@ const greeting = () => {
...
+  showMessage(`Great choice! I love that song ❤ Do you want to play it now or later`, 'bot')
+  setTimeout(() => greatChoice(value, 'bot'), 2000)
+  inputWrapper.innerHTML = `
+  <button id="now">NOW</button>
+  <button id="later">LATER</button>
+  `
+   document.getElementById("now").addEventListener("click", () => {
+        showMessage("Ok, let's listen to it right away! Enjoy your song! Goodbye!", "bot")
+        inpt = () => {
+        }
```
Comment (KarolCislowski) [inline code/script.js:144]: I think you miss something here 🗺️

File: code/script.js [hunk 4]
```diff
@@ -36,12 +44,183 @@ const greeting = () => {
...
+form.addEventListener('submit', (event) => {
+  event.preventDefault();
+  const value = document.getElementById('name-input').value;
+  showMessage(value, 'user')
+});
+
+form.addEventListener('submit', (event) => {
+  event.preventDefault();
+  showMessage(value, 'bot')
+});
```
Comment (KarolCislowski) [inline code/script.js]: You created 3 listeners for the same action and object and this is probably why your inputs do not always show correctly.

File: code/script.js [hunk 5]
```diff
@@ -36,12 +44,183 @@ const greeting = () => {
...
+  event.preventDefault();
+  showMessage(value, 'bot')
+  setTimeout (() => greatChoice(value, 'bot'), 1000)
+});
+
+form.addEventListener('change', (event) => {
+  event.preventDefault();
+  showMessage(value, 'bot')
+  setTimeout (() => whenToplaySong(value, 'bot'), 1000)
+});
```
Comment (KarolCislowski) [inline code/script.js]: again here you have 2 listeners for the same object and action.

File: code/script.js [hunk 6]
```diff
@@ -36,12 +44,183 @@ const greeting = () => {
...
+  <button id="later">LATER</button>
+`
+  document.getElementById("now");
+  showMessage("Ok, let's listen to it right away! Enjoy your song! Goodbye!", "bot")
+
+  document.getElementById("later");
+  showMessage("Later it is! Enjoy your song! Goodbye!", "bot")
+  } 
+
+*/
```
Comment (KarolCislowski) [inline code/script.js]: remember to always remove commented out code 🤓

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
Hi, I'm Karol Cislowski and I'm Technigo's teacher assistant. Sometimes it will be me providing Code Reviews for your assignments. If you have any questions or want to chat about code then feel free to reach me on Technigo's Slack. So let's come back to our business 😸 

You built a nice app here, but you need to clean up it a little bit 🌈 
Try to fix the issue with your inputs. They randomly decide to change sometimes and creates situations like on the screenshot below:


chatbot wrong inputs

File: code/script.js (hunk 2 above)
Comment (esteficodes) [reply code/script.js:125]: Can you leave an example on how to do that? I'm not entirely sure of what you mean.

**Review by esteficodes — COMMENTED** [review]
We are done with the requested changes. Can you take a look at them to see if the project is complete now? 😃😃 @KarolCislowski @HIPPIEKICK

**Review by KarolCislowski — APPROVED** [review]
Great Job!

**Pull request closed without merging**